### ISSUE TITLE
Multiple issues surrounding ZIPprofiles support

### ISSUE THREAD

**Issue by james-stahr** [phaag/nfsen#25]
1) nfdump_version isn't being set properly.
# /data/nfsen/bin/nfsen start
Starting nfcapd:(fit_2)Use of uninitialized value $nfdump_version in numeric eq (==) at /data/nfsen/libexec/NfSenRC.pm line 89.
Use of uninitialized value $nfdump_version in numeric eq (==) at /data/nfsen/libexec/NfSenRC.pm line 122.

2)  nfcapd 1.6.23 doesn't support ZIPprofiles = "-z=lz4" - the  equal sign is not allowed
# /usr/bin/nfcapd -V
/usr/bin/nfcapd: Version: 1.6.23

I enabled the Run statement to get the execution print out, here's the error when starting nfcapd:

Run: /usr/bin/nfcapd -D -p 9996 -u www-data -g www-data -B 500000 -S 1 -P /data/nfsen/var/run/p9996.pid -z=lz4 -T all -t 300 -I fit_2 -l /data/nfsen/profiles-data/live/fit_2  
/usr/bin/nfcapd: invalid option -- '='
usage /usr/bin/nfcapd [options] 
-h              this text you see right here
-u userid       Change user to username
-g groupid      Change group to groupname
-w              Sync file rotation with next 5min (default) interval
-t interval     set the interval to rotate nfcapd files
-b host         bind socket to host/IP addr
-J mcastgroup   Join multicast group <mcastgroup>
-p portnum      listen on port portnum
-l basdir       set the output directory. (no default) 
-S subdir       Sub directory format. see nfcapd(1) for format
-I Ident        set the ident string for stat file. (default 'none')
-H Add port histogram data to flow file.(default 'no')
-n Ident,IP,logdir      Add this flow source - multiple streams
-N sourceFile   Add flows from sourceFile
-M dir          Set the output directory for dynamic sources.
-P pidfile      set the PID file
-R IP[/port]    Repeat incoming packets to IP address/port. Max 8 repeaters.
-s rate set default sampling rate (default 1)
-x process      launch process after a new file becomes available
-z              LZO compress flows in output file.
-y              LZ4 compress flows in output file.
-j              BZ2 compress flows in output file.
-B bufflen      Set socket buffer to bufflen bytes
-e              Expire data at each cycle.
-D              Fork to background
-E              Print extended format of netflow data. For debugging purpose only.
-T              Include extension tags in records.
-4              Listen on IPv4 (default).
-6              Listen on IPv6.
-V              Print version and exit.
-Z              Add timezone offset to filename.
: collector did not start - see logfile (fit_1)

Further,  ZIPprofiles isn't even being set to lz4.  

root@h-nflow-fit-1:/data/nfsen# egrep ZIPprofiles etc/nfsen.conf 
#$ZIPprofiles    = 1;
#$ZIPprofiles   = "-j";
$ZIPprofiles     = "-z=bz2";
#$BZIPprofiles   = 1;

3)  I built up nfdump-1.7.3 which does support -z=, nfcapd does start but not with bz2:

# /data/nfsen/bin/nfsen start
Starting nfcapd:(fit_1)Use of uninitialized value $nfdump_version in numeric eq (==) at /data/nfsen/libexec/NfSenRC.pm line 89.
Use of uninitialized value $nfdump_version in numeric eq (==) at /data/nfsen/libexec/NfSenRC.pm line 122.

Run: /usr/local/bin/nfcapd -D -p 9995 -u www-data -g www-data -B 200000 -S 1 -P /data/nfsen/var/run/p9995.pid -z=lz4  -t 300 -I fit_1 -l /data/nfsen/profiles-data/live/fit_1  
-l is a legacy option and may get removed in future. Please use -w to set output directory
[1028886] (fit_2)Use of uninitialized value $nfdump_version in numeric eq (==) at /data/nfsen/libexec/NfSenRC.pm line 89.
Use of uninitialized value $nfdump_version in numeric eq (==) at /data/nfsen/libexec/NfSenRC.pm line 122.

Run: /usr/local/bin/nfcapd -D -p 9996 -u www-data -g www-data -B 200000 -S 1 -P /data/nfsen/var/run/p9996.pid -z=lz4  -t 300 -I fit_2 -l /data/nfsen/profiles-data/live/fit_2  
-l is a legacy option and may get removed in future. Please use -w to set output directory
[1028893] 
Starting nfsend.

So is the parsing of my nfsen.conf the issue - something tainted in the importation which is causing everything else to be screwed up?  Possibly, because if I manually set libexec/NfSenRC.pm line 401 to  my $nfdump_version = "7";

the startup errors go away, however bz2 is still not honored:

www-data 1028977  0.0  0.0  41292  1944 ?        Sl   13:28   0:00 /usr/local/bin/nfcapd -D -p 9995 -u www-data -g www-data -B 200000 -S 1 -P /data/nfsen/var/run/p9995.pid -z=lz4 -t 300 -I fit_1 -w /data/nfsen/profiles-data/live/fit_1
www-data 1028984  0.0  0.0  41292  1964 ?        Sl   13:28   0:00 /usr/local/bin/nfcapd -D -p 9996 -u www-data -g www-data -B 200000 -S 1 -P /data/nfsen/var/run/p9996.pid -z=lz4 -t 300 -I fit_2 -w /data/nfsen/profiles-data/live/fit_2
root     1029004  0.0  0.0   6608  2296 pts/1    S+   13:29   0:00 grep --color=auto nfcapd
root@h-nflow-fit-1:/data/nfsen/etc# egrep ZIP nfsen.conf
$ZIPcollected    = 1;
#$ZIPprofiles    = "-z=lz4";
$ZIPprofiles     = "-z=bz2";

OS: Ubuntu 22.04.3 LTS
Ubuntu nfdump 1.6.23 and compiled 1.7.3
nfsen version 1.3.11

since it's perl I can hack things to make the default be bz2, but that won't help anyone else.  Let me know if you need anything else

**Comment by stahr** [comment]
okay,  i think I might need to start over as I now see that I was adjusting ZIPprofiles when I should have been adjusting ZIPcollected.  I interpreted the ZIPcollected as a boolean compression setting and the later as the preferred profile.  

But now when I changed ZIPcollected=1 to ZIPcollected="-z=bz2", I got the right result.  

I'm also going to review one of the other closed cases in regards to the version issue..

**Comment by stahr** [comment]
the version issue with the Ubuntu versions is likely due the same issue as #3 as it has NSEL-NEL in the output:

# /usr/bin/nfdump -V
/usr/bin/nfdump: Version: NSEL-NEL1.6.23

however I get the same thing when trying to use the 1.7.3 build when I don't explicitly set $nfdump_version = "7";

# /usr/local/bin/nfdump -V
/usr/local/bin/nfdump: Version: 1.7.3-9a50550 Options: NSEL-NEL Date: 2023-10-08 11:39:43 +0200

however...  nfsen doesn't seem to be getting its' own version either:

# /data/nfsen/bin/nfsen status
Use of uninitialized value in concatenation (.) or string at /data/nfsen/libexec/NfSenRC.pm line 310.
Version nfSen: -1, nfdump: 
NfSen status:
Collector for (fit_1) port 9995 is running [1031970].
Collector for (fit_2) port 9996 is running [1031963].
nfsen daemon:  pid: [1031976] is running.

**Comment by phaag** [comment]
The current main repo needs to be fixed, such that nfsen refuses to work with nfdump-1.6.x. Nfdump-1.6.x is now deprecated and should be replace with 1.7.x If you still want to use nfdump-1.6.x. use nfsen-1.3.9 as latest release. Newer versions are incompatible.
If you upgrade to nfdump-1.7.x do not forget to reinstall NfSen over the current installation as the installation process will check nfdump.

**Comment by stahr** [comment]
Okay.  I wasn't getting that result just now but after doing a re-install I am.  No other changes appear to have been made to the files installed so I can only assume that the above information is only stored/maintained in the profiles-stat/hints file.  So I think we can mark this as resolved as the prior "install" would have been done with PREFIX defined as /usr/bin, which would have been pointing at a nfdump-1.6.23 install.

The backstory here is that I was moving data from an old system to a new one (CentOS 7/nfdump 1.6.22/nfsen 1.3.8) *and* had my sources named using a dash rather than an underscore.   Well, I found this out the hard way that when one "reconfigures" a source, the prior one gets erased so I found myself having to copy the data several times because it kept getting removed.  I'd tried several tricks and even copied all of the files to new directories but in the process of troubleshooting I managed to lose *both* copies, which prompted me to open a report.  

I finally got around it today by just rsyncing the nfcapd files  and the rrd's.  In hindsight, I suspect that copying the hints and profile.dat files were what was tripping me up.    That and thinking that nfdump-1.6 would work with nfsen-1.3.11 when I ran install.pl
